#158487968 Implement App Tour for Players

iverenshaguy opened the pull request:
#### What does this PR do?
This pull request implements the gaming page tour for new players

#### Description of Task to be completed?
- implement app tour of gaming page
- change notifications to toaster

#### How should this be manually tested?
N/A

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
 #158487968

#### Screenshots (if appropriate)
<img width="1440" alt="screen shot 2018-07-17 at 5 26 56 pm" src="https://user-images.githubusercontent.com/22734663/42831592-a679cc04-89e6-11e8-9d57-68b67dd34f22.png">

<img width="1440" alt="screen shot 2018-07-18 at 10 46 04 am" src="https://user-images.githubusercontent.com/22734663/42873906-d72c1442-8a77-11e8-8b30-248a9d9586b4.png">


#### Questions:
N/A

otseobande commented on client/js/controllers/game.js:
```diff
@@ -304,8 +303,6 @@ angular.module('mean.system')
           Global.getUser().then(function (response) {
             if (response.authenticated) {
-              console.log(response.authenticated);
-              console.log(donation);
```
Good job removing the `console.log` statements.

otseobande commented on client/views/app.html:
```diff
@@ -30,7 +30,7 @@
           </div>
         </li>
-        <li><a class="waves-effect waves-yellow btn tour-btn">Take A Tour</a></li>
+        <li><a class="waves-effect waves-yellow btn tour-btn" ng-click="CallMe()">Take A Tour</a></li>
```
`callMe()` is definitely a funny function name 🙂. Is there any reason it's named that way?

iverenshaguy replied:
It's not my function, it's from the angular-intro plugin. You can go to their plugin page to research why they decided to call their function CallMe(). https://github.com/mendhak/angular-intro.js

otseobande replied:
Alright

otseobande commented on client/views/question.html:
```diff
@@ -1,35 +1,38 @@
     </div>
-    <p class="text-center">{{game.players.length}}&nbsp;/&nbsp;12 Players Found</p>
+  </div>
+    <p class="text-center" id="players-found-text">{{game.players.length}}&nbsp;/&nbsp;12 Players Found</p>
```
I think we could use other techniques to add whitespaces other than`&nbsp;`. We could use the `<pre>` tag for retaining the format in which we wrote the text in or use a `white-space` CSS property to achieve the same effect. `&nbsp;` has become one of the various HTML artifacts. I feel we should join the evolution of the web and use these other techniques.

iverenshaguy replied:
Ok

iverenshaguy replied:
`<pre>` changed it to a monospaced font

iverenshaguy replied:
The `white-space` property also didn't work. However, I was able to remove the nbsp; and leave a space between and it maintained the space just like a normal sentence.

otseobande commented on client/views/tourview.html:
```diff
@@ -423,17 +359,9 @@
-</body>
-</html>
\ No newline at end of file
+      </div>
```
Just like Victoria would always note. We missed an extra line here.

iverenshaguy replied:
We? I missed an extra line here. Fixing it now. Thanks

otseobande reviewed (commented):
You implemented this feature impressively. However, there are somethings that I still need to clarify. I noticed that the element attribute of an object in the steps array in the TourController uses an id selector. This makes sense as ids are a way to mark unique elements but this has led to us having ids and classes with the same names on the elements which the stepper is implemented on. Do you think its a good idea if we remove the class names and convert the styles to select the elements using these ids instead?

iverenshaguy commented:
I always use classes to style elements so that the modules are reusable. I used IDs for the angular-intro plugin because if I were to use classes, I'd need to append the position of each element as in ('.elem')[0] and I didn't want to get confused. I will change the id names to unique names so that they are not the same as the class names.

Stylll reviewed (approved):
It would have been nice if we had a little demo of the tour in form of a gif in your PR rather than just a screenshot. Nevertheless, you've done a good job. Keep it up.

iverenshaguy committed:
feat(app-tour): implement app tour for new users

- implement app tour of gaming page
- change notifications to toaster

[Finishes #158487968]
